pkg/cluster/arooperator.go

```diff
@@ -29,23 +30,24 @@ func (m *manager) ensureAROOperator(ctx context.Context) error {
 
 func (m *manager) aroDeploymentReady(ctx context.Context) (bool, error) {
 	if !m.isIngressProfileAvailable() {
+		// If the ingress profile is not available, ARO operator update/deploy will fail.
 		m.log.Error("skip aroDeploymentReady")
 		// skip and don't retry
 		return true, nil
 	}
 	return m.aroOperatorDeployer.IsReady(ctx)
 }
 
-func (m *manager) ensureAROOperatorRunningDesiredVersion(ctx context.Context) error {
+func (m *manager) ensureAROOperatorRunningDesiredVersion(ctx context.Context) (bool, error) {
 	if !m.isIngressProfileAvailable() {
+		// If the ingress profile is not available, ARO operator update/deploy will fail.
 		m.log.Error("skip ensureAROOperatorRunningDesiredVersion")
-		return nil
+		// skip and don't retry
```

Comment:
No need for this comment as the log error message says it's going to skip it.  A bit redundant.